frontend/src/util/session/report.ts

```diff
@@ -0,0 +1,125 @@
+import {
+	useGetSessionsClickhouseLazyQuery,
+	useGetSessionsReportLazyQuery,
+} from '@graph/hooks'
+import { ClickhouseQuery, Session, SessionsReportRow } from '@graph/schemas'
+import { useProjectId } from '@hooks/useProjectId'
+import { useSearchContext } from '@pages/Sessions/SearchContext/SearchContext'
+
+const getQueryRows = (query: ClickhouseQuery, sessions: Session[]) => [
+	['filters', 'num_results'],
+	[query.rules.map((rule) => rule.join(':')).join('|'), sessions.length],
+]
+
+const getReportRows = (sessionsReportRows: SessionsReportRow[]) => {
+	return [
+		Object.keys(sessionsReportRows[0]).filter(
+			(key) => key !== '__typename',
+		),
+		...sessionsReportRows.map((report: SessionsReportRow) => {
+			return Object.values(report).filter(
+				(value) => value !== 'SessionsReportRow',
+			)
+		}),
+	]
+}
+
+const getSessionRows = (sessions: Session[]) => {
+	const rows = []
+	for (const session of sessions) {
```

Comment:
Maybe an improvement for the future, but think it would be nice to only have the headers once instead of per row

Reply:
yeah, the customer actually brought this up (i didn't implement it that way since rows can have different keys), so i'll be aggregating keys for one header and leaving unset ones blank